### ISSUE TITLE
Literals over JS's `Number.MAX_SAFE_INTEGER` are incorrect

### ISSUE THREAD

**Issue by bluetech** [microsoft/pyright#2879]
**Describe the bug**

I haven't checked, but I guess that pyright represent `Literal` `int`'s internally using JS's `Number` type, however `Number` is not arbitrary-precision like Python's `int` which causes overflow problems.

**To Reproduce**

```py
reveal_type(9007199254740993)
# Type of "9007199254740992" is "Literal[9007199254740992]"
```

The magic number is `BigInt(Number.MAX_SAFE_INTEGER) + 2n` (in JS), i.e. `2^53 + 1`. See [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER). This happens to (most) numbers above this limit (and the negatives).

**Expected behavior**

Revealed type is `Literal[9007199254740993]`.

**VS Code extension or command-line**

```con
$ ./node_modules/.bin/pyright --version
pyright 1.1.210
```

**Additional context**

Tried it after seeing the literal math discussion. I think Python type checkers need to think twice before adding literal math - it's a semantic mine field, and the fact that TypeScript (which allows some amazing things with literal types) doesn't have it gives me pause... Though this bug is not directly really related to that, it illustrates the kind of issues that might pop up.

For fixing the bug, perhaps pyright can use `bigint` instead of `number` to represent Literal `int`s. According to MDN bigint is available since node 10.4.0.

**Comment by JelleZijlstra** [comment]
This is a duplicate of #2860.

**Comment by erictraut** [comment]
Have you hit this in actual code, or is this just a theoretical concern?

**Comment by bluetech** [comment]
Oops, didn't notice the closed one. It is theoretical.

**Comment by erictraut** [comment]
Yeah, this is a known issue.

**Comment by erictraut** [comment]
I had planned to wait until this became more than just theoretical, but since you're the third person to mention it in the last few days, I've added support for bigint values in the tokenizer and the type system, including the new literal math logic.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.211, which I just published. It will also be included in the next release of pylance.
